enterprise/server/remote_execution/containers/podman/podman.go

```diff
@@ -112,13 +123,24 @@ additionallayerstores=["/var/lib/soci-store/store:ref"]
 			return nil, status.UnavailableErrorf("could not write storage config: %s", err)
 		}
 
-		// To prevent podman trying to read from the soci-store directory
-		// before it's ready (which can take a few hundred ms), wait for up to
-		// one second for soci-store to start up. Note that this is happening
-		// during executor start up so it won't affect RBE times.
-		if err := disk.WaitUntilExists(context.Background(), sociStoreDir, disk.WaitOpts{}); err != nil {
+		if err := disk.WaitUntilExists(context.Background(), sociStorePath, disk.WaitOpts{}); err != nil {
 			return nil, status.UnavailableErrorf("soci-store failed to start: %s", err)
 		}
+
+		// TODO(iain): there's a concurrency bug in soci-store that causes it
+		// to die occasionally. Report the executor as unhealthy if the
+		// soci-store directory doesn't exist for any reason.
+		env.GetHealthChecker().AddHealthCheck(
+			"soci_store", interfaces.CheckerFunc(
+				func(ctx context.Context) error {
+					if _, err := os.Stat(sociStorePath); err == nil {
+						return nil
+					} else {
```

Comment:
nit: unbalanced parenthesis in error message

Reply:
Fixed